smithy-aws-protocol-tests/model/restJson1/http-content-type.smithy

```diff
@@ -0,0 +1,293 @@
+            "Content-Type": "application/json"
+        },
+        requireHeaders: [
+            "Content-Length"
+        ],
+        params: {
+            testId: "t-12345"
+        }
+    }
+])
+
+structure TestPayloadStructureInputOutput {
+    @httpHeader("x-amz-test-id")
+    testId: String,
+
+    @httpPayload
+    payloadConfig: PayloadConfig
+}
+
+structure PayloadConfig {
+    data: Integer
+}
+
+/// This example operation serializes a payload targeting a blob.
+///
+/// The Blob shape is not structured content and we cannot
+/// make assumptions about what data will be sent. This test ensures
+/// only a generic "Content-Type: application/octet-stream" header
+/// is used, and that we are not treating an empty body as an
+/// empty JSON document.
+///
+@idempotent
+@http(uri: "/blob_payload", method: "POST")
+operation TestPayloadBlob {
+    input: TestPayloadBlobInputOutput,
+    output: TestPayloadBlobInputOutput
+}
+
+apply TestPayloadBlob @httpRequestTests([
+    {
+        id: "RestJsonHttpWithEmptyBlobPayload",
+        documentation: "Serializes a payload targeting an empty blob",
+        protocol: restJson1,
+        method: "POST",
+        uri: "/blob_payload",
+        body: "",
+        bodyMediaType: "application/octet-stream",
+        headers: {
+            "Content-Type": "application/octet-stream"
+        },
+        requireHeaders: [
+            "Content-Length"
+        ],
+        params: {}
+    }
+])
+
+apply TestPayloadBlob @httpRequestTests([
+    {
+        id: "RestJsonTestPayloadBlob",
+        documentation: "Serializes a payload targeting a blob",
+        protocol: restJson1,
+        method: "POST",
+        uri: "/payload_blob",
+        body: """
+              1234
+              """,
+        bodyMediaType: "image/jpg",
+        headers: {
+            "Content-Type": "image/jpg"
+        },
+        requireHeaders: [
+            "Content-Length"
+        ],
+        params: {
+            contentType: "image/jpg",
+            data: "1234"
+        }
+    }
+])
+
+structure TestPayloadBlobInputOutput {
+    @httpHeader("Content-Type")
+    contentType: String,
+
+    @httpPayload
+    data: Blob
+}
+
+/// This example operation serializes a request without an HTTP body.
+///
+/// These tests are to ensure we do not attach a body or related headers
+/// (Content-Length, Content-Type) to operations that semantically
+/// cannot produce an HTTP body.
+///
+@idempotent
+@http(uri: "/no_payload", method: "GET")
+operation TestNoPayload {
+    input: TestNoPayloadInputOutput,
+    output: TestNoPayloadInputOutput
+}
+
+apply TestNoPayload @httpRequestTests([
+    {
+        id: "RestJsonHttpWithNoModeledBody",
+        documentation: "Serializes a GET request with no modeled body",
+        protocol: restJson1,
+        method: "GET",
+        uri: "/no_payload",
+        body: "",
+        headers: {
+        },
+        requireHeaders: [
+            "Content-Length",
+            "Content-Type"
```

Comment:
Why is `Content-Type` required here?

Reply:
These TestNoPayload tests should both be using `forbidHeaders` not `requireHeaders`.

Reply:
wait, it `Content-Length` actually forbidden in this case? Why can't `Content-Length: 0` be sent here?

Reply:
We're being slightly overly prescriptive here but this is the behavior we're intending for the SDKs. 

>A user agent SHOULD NOT send a Content-Length header field when the request message does not contain
   a payload body and the method semantics do not anticipate such a body.
   
([RFC 7230 § 3.3.2](https://datatracker.ietf.org/doc/html/rfc7230#section-3.3.2))
   
   This was left as SHOULD NOT because there are a few semi-common implementations with this behavior. It's broadly considered incorrect though and the goal is for new clients we're creating to do the right thing by default.